schema/synthetix/view_trades.sql

```diff
@@ -1,55 +1,35 @@
+)
+SELECT
     trade.evt_block_time AS block_time,
     'Synthetix' AS project,
     '1' AS version,
     trade.account AS trader_a,
     NULL::bytea AS trader_b,
     trade."fromAmount"/1e18 AS token_a_amount_raw,
     trade."toAmount"/1e18 AS token_b_amount_raw,
-    synths_a.address AS token_a_address,
-    synths_b.address AS token_b_address,
+    (SELECT address FROM synthetix.view_synths WHERE trade."fromCurrencyKey" = currency_key AND evt_block_time < trade.evt_block_time ORDER BY evt_block_time DESC LIMIT 1) AS token_a_address,        
+    (SELECT address FROM synthetix.view_synths WHERE trade."toCurrencyKey" = currency_key AND evt_block_time < trade.evt_block_time ORDER BY evt_block_time DESC LIMIT 1) AS token_b_address,
     trade.contract_address AS exchange_contract_address,
     trade.evt_tx_hash AS tx_hash,
     NULL::integer[] AS trace_address,
     trade.evt_index AS evt_index,
-    trade."fromAmount"/1e18 * rates_a.currency_rate/1e18 as token_a_amount,
-    trade."toAmount"/1e18 * rates_a.currency_rate/1e18 as token_b_amount,
-    rates_a.currency_rate/1e18 as token_a_usd_price,
-    rates_b.currency_rate/1e18 as token_b_usd_price
+    trade."fromAmount"/1e18 * (SELECT currency_rate FROM rates WHERE trade."fromCurrencyKey" = currency_key AND evt_block_time < trade.evt_block_time ORDER BY evt_block_time DESC LIMIT 1)/1e18 as token_a_amount,
+    trade."toAmount"/1e18 * (SELECT currency_rate FROM rates WHERE trade."toCurrencyKey" = currency_key AND evt_block_time < trade.evt_block_time ORDER BY evt_block_time DESC LIMIT 1)/1e18 as token_b_amount
 FROM
-    synthetix."Synthetix_evt_SynthExchange" trade
-LEFT JOIN synthetix."view_synths" synths_a
-    ON trade."fromCurrencyKey" = synths_a.currency_key
-    AND trade.evt_block_time >= synths_a.evt_block_time
-    AND trade.evt_block_time < synths_a.max_block_time
-LEFT JOIN synthetix."view_synths" synths_b
-    ON trade."toCurrencyKey" = synths_b.currency_key
-    AND trade.evt_block_time >= synths_b.evt_block_time
-    AND trade.evt_block_time < synths_b.max_block_time
-LEFT JOIN synthetix."view_synths_rates" rates_a
-    ON trade."fromCurrencyKey" = rates_a.currency_key
-    AND trade.evt_block_time >= rates_a.evt_block_time
-    AND trade.evt_block_time < rates_a.max_block_time
-LEFT JOIN synthetix."view_synths_rates" rates_b
-    ON trade."toCurrencyKey" = rates_b.currency_key
-    AND trade.evt_block_time >= rates_b.evt_block_time
-    AND trade.evt_block_time < rates_b.max_block_time
-;
+    synthetix."Synthetix_evt_SynthExchange" trade;
+
+CREATE UNIQUE INDEX IF NOT EXISTS view_trades_id ON synthetix.view_trades (block_time,evt_index);
+CREATE INDEX view_trades_block_time ON synthetix.view_trades (block_time);
+CREATE INDEX view_trades_token_a_address ON synthetix.view_trades (token_a_address);
+CREATE INDEX view_trades_token_b_address ON synthetix.view_trades (token_b_address);
+CREATE INDEX view_trades_token_a_amount ON synthetix.view_trades (token_a_amount);
```

Comment:
I can't think of any use for this index, so unless there's a compelling case I haven't considered, we could remove it